### DISCUSSION TITLE
[Question][Hw5] Cmp in Taint domain

### DISCUSSION THREAD

**Discussion by KAIST-JongchanPark** [prosyslab-classroom/cs348-information-security#619] [Q&A]
In Taint Domain, there is a function `let cmp _ = failwith "Not implemented"`

1. Do I need to change it into `let cmp pred v1 v2 = implementation `, similar to Sign Domain?

-------------------------------------

```
a = 3
b = source() 
c = a < b
if (a < b) {  
  sink(a)  
}
```
For above case, before if statement, a is bottom(None) and b is top(taint).

2. In this situation, c would be taint because b is taint? (Question about action of cmp function in taint domain)
3. In this situation, when I filter `a<b`, a become taint because b is taint? (Question about action of filter function in taint domain)

**Comment by john-mai-2605** [comment]
For 3, see #163 and #147

**Comment by RiceBiscuits** [comment]
1. If you needed.
2. if the variable `c` could be affected by variable `a` and `b`, then yes.
